### PR TITLE
Update Azurestack DiskRP admin SDK

### PR DESCRIPTION
**Pull request by shanswmicrosoft**
## Description

Update Azurestack DiskRP admin SDK - compute module. It includes two kinds of resources:
a. Disk - Get and List
b. DiskMigrationJob - Get, List, Create and Stop. 

PR for Swagger: 
https://github.com/Azure/azure-rest-api-specs/pull/3418

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                    {
+                        toMigrationDisks.Add(disk);
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+
+                var migrationId = "ba0644a4-c2ed-4e3c-a167-089a32865297"; // System.Guid.NewGuid().ToString(); This guid should be the same as the ones in sessionRecord
```
Comment (deathly809) [inline src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs]: Remove 'System.Guid.NewGuid().StoString();' from comment.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin/Microsoft.AzureStack.Management.Compute.Admin.csproj [hunk 2]
```diff
@@ -7,13 +7,14 @@
     <PropertyGroup>
         <PackageId>Microsoft.AzureStack.Management.Compute.Admin</PackageId>
         <Description>Provides developers with libraries for the compute admin under AzureStack to manage compute resources.</Description>
-        <VersionPrefix>0.2.0-preview</VersionPrefix>
+        <VersionPrefix>0.3.0-preview</VersionPrefix>
         <AssemblyName>Microsoft.AzureStack.Management.Compute.Admin</AssemblyName>
         <PackageTags>Microsoft Azure Stack;Compute;REST HTTP client;azureostackfficial;netcore451511</PackageTags>
```
Comment (deathly809) [inline src/AzureStack/Admin/ComputeAdmin/Compute.Admin/Microsoft.AzureStack.Management.Compute.Admin.csproj]: You might want to add 'Managed Disks' to the PackageTags.  Add others if you feel necessary.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
+﻿
```
Comment (deathly809) [inline src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs]: Can you do tests that should fail?  For instance try to migrate to a share that doesn't exist or with a disk that is invalid?

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj [hunk 4]
```diff
@@ -6,9 +6,9 @@
     <AssemblyName>Compute.Admin.Tests</AssemblyName>
     <VersionPrefix>1.0.0</VersionPrefix>    
   </PropertyGroup>
-  <!--<PropertyGroup>
```
Comment (dsgouda) [inline src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj]: Please remove this change. Our tests are now running against netcoreapp2.0

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from the minor comment

**Comment by deathly809** [comment]
@dsgouda  We would like tests that verify invalid input.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj (hunk 4 above)
Comment (deathly809) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj]: You are saying they should leave this commented out?  If so they should just delete this XML entry then.

Comment (dsgouda) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj]: Leave it commented for now, we will do a cleanup throughout the repo at a later point.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs (hunk 3 above)
Comment (dsgouda) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs]: Try `Assert.False` or `Assert.Exception` constructs.
If you wish to add the tests but enable them at a later time, you could always skip the tests, hope this helps

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs (hunk 1 above)
Comment (shanswmicrosoft) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs]: sure. thanks.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin/Microsoft.AzureStack.Management.Compute.Admin.csproj (hunk 2 above)
Comment (shanswmicrosoft) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin/Microsoft.AzureStack.Management.Compute.Admin.csproj]: Sure.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj (hunk 4 above)
Comment (shanswmicrosoft) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj]: OK. I will leave it as what it is.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs (hunk 3 above)
Comment (shanswmicrosoft) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/DiskMigrationJobTests.cs]: I have added more tests.

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj (hunk 4 above)
Comment (shanswmicrosoft) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/Compute.Tests.csproj]: @dsgouda I change it back. thanks.

**Comment by dsgouda** [comment]
Will review the rest upon @deathly809 's approval

**Comment by deathly809** [comment]
@dsgouda Looks good to me!  :)

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/Helpers/AzureStackTestBase.cs [hunk 5]
```diff
@@ -85,7 +85,10 @@ public abstract class AzureStackTestBase<T> where T : class
 
         public void ValidateExpectedReturnCode(Action action, HttpStatusCode httpResponseCode)
         {
-            try { action.Invoke(); }
+            try {
```
Comment (dsgouda) [inline src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/Helpers/AzureStackTestBase.cs]: nit: Highly recommend using `Assert.Exception` here. Take a look at [this](https://stackoverflow.com/questions/45017295/assert-an-exception-using-xunit)

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from a minor comment. Won't block if you choose to integrate it later

File: src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/Helpers/AzureStackTestBase.cs (hunk 5 above)
Comment (shanswmicrosoft) [reply src/AzureStack/Admin/ComputeAdmin/Compute.Admin.Tests/src/Helpers/AzureStackTestBase.cs]: Sure. Updated. Thanks.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**